pinot-common/src/main/java/org/apache/pinot/common/utils/config/QueryOptionsUtils.java

```diff
@@ -190,4 +193,18 @@ public static Integer getGroupTrimThreshold(Map<String, String> queryOptions) {
     String groupByTrimThreshold = queryOptions.get(QueryOptionKey.GROUP_TRIM_THRESHOLD);
     return groupByTrimThreshold != null ? Integer.parseInt(groupByTrimThreshold) : null;
   }
+
+  public static boolean isDropResultsEnabled(final JsonNode request) {
+    boolean dropResults = false;
+
+    if (request.has(CommonConstants.Broker.Request.QUERY_OPTIONS)) {
+      String queryOptions = request.get(CommonConstants.Broker.Request.QUERY_OPTIONS).asText();
+      Map<String, String> optionsFromString = RequestUtils.getOptionsFromString(queryOptions);
+      if (Boolean.parseBoolean(optionsFromString.get(CommonConstants.Broker.Request.QueryOptionKey.DROP_RESULTS))) {
+        dropResults = true;
+      }
+    }
+
+    return dropResults;
+  }
```

Comment:
We want to take the `Map` here instead of the `JsonNode`. The logic of extracting the map should be on the caller side
```suggestion
  public static boolean isDropResults(Map<String, String> queryOptions) {
    return Boolean.parseBoolean(queryOptions.get(CommonConstants.Broker.Request.QueryOptionKey.DROP_RESULTS));
  }
```